array.c

```diff
@@ -3126,20 +3128,49 @@ rb_ary_to_a(VALUE ary)
 
 /*
  *  call-seq:
- *     ary.to_h                  -> hash
- *     ary.to_h {|item| block }  -> hash
+ *    array.to_h -> new_hash
+ *    array.to_h {|2_element_array| ... } -> new_hash
```

Comment:
What about `|key, value|` instead of `|2_element_array|`? It would save reading down a bit to learn what the two elements were

Reply:
Aren't both `2 element_array` and `key, value` as block arguments wrong?  The block is yielded each element in the array, and should return a 2 element array.  The previous `item` was more accurate.

Reply:
I think you're right and that I confused myself with the way block parameters destructure the input item:

```
$ ruby -e 'p [["a", 1], ["b", 2]].to_h { |k, v| [k.upcase, v] }'
{"A"=>1, "B"=>2}
$ ruby -e 'p %w[a b].to_h { |e| [e.upcase, e] }'
{"A"=>"a", "B"=>"b"}
```

I agree and think this should be something like `array.to_h {|item| ... } -> new_hash`

Perhaps an example like the last one should be added where an array of single elements is converted to a Hash?

Reply:
Still working this one.  Nuther PR to follow.

Reply:
I have revised the doc for #to_h.